spec/eof.md

```diff
@@ -288,18 +288,15 @@ The following instructions are introduced in EOF code:
 - the first code section must have a type signature `(0, 0x80, max_stack_height)` (0 inputs non-returning function)
 - `EOFCREATE` `initcontainer_index` must be less than `num_container_sections`
 - `EOFCREATE` the subcontainer pointed to by `initcontainer_index` must have its `len(data_section)` equal `data_size`, i.e. data section content is exactly as the size declared in the header (see [Data section lifecycle](#data-section-lifecycle))
-- `EOFCREATE` the subcontainer pointed to by `initcontainer_index` must be an "initcode" subcontainer, that is, it *must not* contain either a `RETURN` or `STOP` instruction.
+- `EOFCREATE` the subcontainer pointed to by `initcontainer_index` *must not* contain either a `RETURN` or `STOP` instruction.
 - `RETURNCONTRACT` `deploy_container_index` must be less than `num_container_sections`
-- `RETURNCONTRACT` the subcontainer pointed to `deploy_container_index` must be a "runtime" subcontainer, that is, it *must not* contain a `RETURNCONTRACT` instruction.
+- `RETURNCONTRACT` the subcontainer pointed to `deploy_container_index` *must not* contain a `RETURNCONTRACT` instruction.
 - `DATALOADN`'s `immediate + 32` must be within `pre_deploy_data_size` (see [Data Section Lifecycle](#data-section-lifecycle))
      - the part of the data section which exceeds these bounds (the `dynamic_aux_data` portion) needs to be accessed using `DATALOAD` or `DATACOPY`
 - no unreachable code sections are allowed, i.e. every code section can be reached from the 0th code section with a series of CALLF / JUMPF instructions, and section 0 is implicitly reachable.
 - it is an error for a container to contain both `RETURNCONTRACT` and either of `RETURN` or `STOP`.
-- it is an error for a subcontainer to never be referenced in code sections of its parent container
-- for terminology purposes, the following concepts are defined:
```

Comment:
I'm leaning to keeping these definitions, I don't feel like they dictate particular implementation, and I expect these concepts to appear in tests (at least as "initcode container" flag to mark some tests, "runtime" should be default and might not be mentioned).

But it would make sense to move the definitions to above their first use.

Reply:
> I expect these concepts to appear in tests (at least as "initcode container" flag to mark some tests

Just got an idea - this flag could just as well be "creation_tx_container", as it would be what it is in production. Then the default (ex. "runtime container"), would be much more natural. WDYT?

Reply:
However, actually, `initcode container` or `initcontainer` is fine as a testing flag, even if these definitions are not here. The rule is in there in cases where it applies (creation tx, txcreate, eofcreate), and all of these name this "thing" as `initcontainer`, so I think it is all clear enough.